psga/common/primer_autodetection.nf

```diff
@@ -0,0 +1,81 @@
+/*
+ * Run: primer-autodetection (use first read only for illumina)
+ */
+process primer_autodetection {
+  publishDir "${params.output_path}/primer_autodetection", mode: 'copy', overwrite: true, pattern: '{*_primer.txt,*_primer_data.csv,*_primer_detection.tsv,trimmomatic.out,*.bowtie2*}'
+
+  tag "${task.index} - ${file_1}"
+
+  input:
+    path file_1
+    val kit
+
+  output:
+    path "*_primer.txt", emit: ch_primer
+    path "*_primer_data.csv", emit: ch_primer_data
+    path "*_primer_detection.tsv", emit: ch_primer_coverage
+    path "trimmomatic.out"
+    path "*.bowtie2*"
+
+  shell:
+  '''
+  kit=!{kit}
+  file_1=!{file_1}
+  sample_id=$( echo ${file_1} | cut -d '.' -f1 )
+  trimmed_sample="trimmed_sample.fastq"
+  crop="!{params.primer_minimum_length}"
+
+  # trim sample reads so that they are long as the shortest primer
+  java -jar /opt/trimmomatic/trimmomatic.jar SE -phred33 ${file_1} ${trimmed_sample} CROP:${crop} 2> trimmomatic.out
+
+  # iterate over the available primer scheme fasta.
+  # These fasta are generated by us and contain the sequences for each primer scheme name/version
+  for primer_fasta in $(find /primer_schemes -iname SARS-CoV-2.scheme.fasta); do
```

Comment:
This is quite an inefficient way of finding this, it would be better to just have a way of knowing where to look.

Reply:
The primers are stored in paths and this structure depends on artic.
We could have a file listing these paths, but in general, this is not slow as there are very few files in those paths.